MockKitUI/MockSupport.swift

```diff
@@ -0,0 +1,144 @@
+        return TimeInterval.minutes(1)
+    }
+
+    private func maybeIssueAlert(_ versionUpdate: VersionUpdate) {
+        guard versionUpdate >= .recommended else {
+            noAlertNecessary()
+            return
+        }
+        
+        let alertIdentifier = Alert.Identifier(managerIdentifier: MockSupport.supportIdentifier, alertIdentifier: versionUpdate.rawValue)
+        let alertContent: LoopKit.Alert.Content
+        if firstAlert {
+            alertContent = Alert.Content(title: versionUpdate.localizedDescription,
+                                         body: NSLocalizedString("""
+                                                Your app is out of date. It will continue to work, but we recommend updating to the latest version.
+                                                """, comment: "Alert content body for first software update alert"),
+                                         acknowledgeActionButtonLabel: NSLocalizedString("OK", comment: "default acknowledgement"),
+                                         isCritical: versionUpdate == .required)
+        } else if let lastVersionCheckAlertDate = lastVersionCheckAlertDate,
+                  abs(lastVersionCheckAlertDate.timeIntervalSinceNow) > alertCadence {
+            alertContent = Alert.Content(title: NSLocalizedString("Update Reminder", comment: "Recurring software update alert title"),
+                                         body: NSLocalizedString("""
+                                                Your app is still out of date. It will continue to work, but we recommend updating to the latest version.
+                                                """, comment: "Alert content body for recurring software update alert"),
+                                         acknowledgeActionButtonLabel: NSLocalizedString("OK", comment: "default acknowledgement"),
+                                         isCritical: versionUpdate == .required)
+        } else {
+            return
+        }
+        alertIssuer!.issueAlert(Alert(identifier: alertIdentifier, foregroundContent: alertContent, backgroundContent: alertContent, trigger: .immediate))
+        recordLastAlertDate()
+    }
+    
+    private func noAlertNecessary() {
+        lastVersionCheckAlertDate = nil
+    }
+    
+    private var firstAlert: Bool {
+        return lastVersionCheckAlertDate == nil
+    }
+    
+    private func recordLastAlertDate() {
+        lastVersionCheckAlertDate = Date()
+    }
+    
+}
+
+struct SupportMenuItem : View {
+    
+    let mockSupport: MockSupport
+    
+    @State var showActionSheet: Bool = false
+    
+    private var buttons: [ActionSheet.Button] {
+        VersionUpdate.allCases.map { versionUpdate in
+            let setter = { mockSupport.versionUpdate = versionUpdate }
+            switch versionUpdate {
+            case .required: return ActionSheet.Button.destructive(Text(versionUpdate.localizedDescription), action: setter)
```

Comment:
I'm am not a fan of having the statement on the same line as the case (I find it much easier to miss the statement being executed), but I can deal with it when it is trivial code. This, however, is not trivial, so please move to separate lines. And, really, what exactly are you saving here by putting them on the same line?